### PR TITLE
(fix) Refactor compact patient search

### PR DESCRIPTION
**Pull request by denniskigen**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing.en-US#contributing-guidelines) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://om.rs/o3ui).
- [x] My work includes tests or is validated by existing tests.

## Summary

This PR refactors the following aspects of the Compact Patient Search:

- Renaming the `PatientSearchResults` component to `CompactPatientBanner`. I don't know why a component in a file named `compact-patient-banner.component.tsx` would be named `PatientSearchResults`, especially when there's another completely different component with the same exact name. I thought I'd done this in #977 but it looks like I didn't include those changes in my diff.
- Renames the `RecentPatientSearch` component to `RecentlySearchedPatients`, a name that better reflects what it does.
- Fixes to various tests so they actually test what they claim to.

### CODE REVIEW

File: packages/esm-patient-search-app/src/compact-patient-search/compact-patient-banner.component.tsx [hunk 1]
```diff
@@ -162,35 +176,31 @@ const ClickablePatientContainer = ({ patient, children }: ClickablePatientContai
...
+const IdentifierTag: React.FC<IdentifierTagProps> = ({ identifier }) => {
+  return (
+    <>
+      <Tag size="sm" className={styles.configuredTag} type="warm-gray" title={identifier.identifierType.display}>
+        {identifier.identifierType.display}
+      </Tag>
+      <span className={styles.configuredLabel}>{identifier.identifier}</span>
+    </>
+  );
+};
```
Comment (denniskigen) [inline packages/esm-patient-search-app/src/compact-patient-search/compact-patient-banner.component.tsx:188]: I've just factored out this shared Tag logic into this component.

File: packages/esm-patient-search-app/src/config-schema.ts [hunk 2]
```diff
@@ -41,20 +41,26 @@ export const configSchema = {
     _elements: {
       _type: Type.UUID,
     },
-    _description: 'The identifier types to be display on all patient search result page',
+    _description:
+      'A list of identifier types to be displayed in the patient search results as banner tags. If no defaultIdentifierTypes are provided, the defaultIdentifier will be displayed.',
+    // These values correspond to the Patient Clinic Number and National Unique Patient Identifier (NUPI) identifier types respectively
```
Comment (brandones) [inline packages/esm-patient-search-app/src/config-schema.ts:46]: thank you, thank you, thank you

**Review by brandones — APPROVED** [review]
LGTM

File: packages/esm-patient-search-app/src/config-schema.ts [hunk 3]
```diff
@@ -41,20 +41,26 @@ export const configSchema = {
...
+    _description: 'Identifer type to be displayed when no defaultIdentifierTypes are provided. Default is OpenMRS ID.',
     _default: 'OpenMRS ID',
   },
 };
 
 export type PatientSearchConfig = {
-  search: { patientSearchResult: string; showRecentlySearchedPatients: string; disableTabletSearchOnKeyUp: boolean };
-  includeDead: boolean;
+  search: {
+    disableTabletSearchOnKeyUp: boolean;
```
Comment (dkayiwa) [inline packages/esm-patient-search-app/src/config-schema.ts:58]: Is disableTabletSearchOnKeyUp used only for tablets? Or can it be also used in other cases?

Comment (ibacher) [reply packages/esm-patient-search-app/src/config-schema.ts:58]: It's in all other cases, but this is just adding the existing configuration item (see line 15 above) to the TS type for the schema.

**Pull request merged**